local-compose.yml

```diff
@@ -1,50 +1,51 @@
-# Local testing version of Elasticsearch stack designed for
-# container-native deployment on Joyent's Triton platform.
+version: '2.1'
+services:
+    # Local testing version of Elasticsearch stack designed for
+    # container-native deployment on Joyent's Triton platform.
+    elasticsearch:
+        build: .
+        environment:
+          - CONSUL_AGENT=1
+          - CONSUL=consul
+          - ES_CLUSTER_NAME=famished_test
+          - ES_SERVICE_NAME=elasticsearch-master
+          - ES_NODE_MASTER=true
+          - ES_NODE_DATA=true
+          - ES_ENVIRONMENT=dev
+          # - ES_CLUSTER_SIZE=1
 
-elasticsearch_master:
-    extends:
-      file: docker-compose.yml
-      service: elasticsearch
-    mem_limit: 512m
-    build: .
-    environment:
-      - CONSUL=consul
-      - ES_SERVICE_NAME=elasticsearch-master
-      - ES_NODE_MASTER=true
-      - ES_NODE_DATA=false
-      - CONTAINERPILOT=file:///etc/containerpilot.json
-    links:
-    - consul:consul
+          # are these even remotely accurate?
+          - ES_JAVA_OPTS=-Djava.security.egd=file:/dev/./urandom -XX:-UseGCTaskAffinity -XX:-BindGCTaskThreadsToCPUs -XX:ParallelGCThreads=1 -XX:ParallelCMSThreads=1 -Xms128m -Xmx128m
 
-elasticsearch:
-    extends:
-      file: docker-compose.yml
-      service: elasticsearch
-    mem_limit: 512m
-    build: .
-    environment:
-      - CONSUL=consul
-      - ES_SERVICE_NAME=elasticsearch-master
-    links:
-      - consul:consul
+          # disable x-pack
+          - xpack.security.enabled=false
 
-elasticsearch_data:
-    extends:
-      file: docker-compose.yml
-      service: elasticsearch
-    mem_limit: 512m
-    build: .
-    environment:
-      - CONSUL=consul
-      - ES_SERVICE_NAME=elasticsearch-data
-      - ES_NODE_MASTER=false
-      - ES_NODE_DATA=true
-    links:
-      - consul:consul
+    elasticsearch_master:
```

Comment:
> I'm not really a big fan of elasticsearch-master, elasticsearch-data and elasticsearch (with a service name of elasticsearch-master). I'm guessing this has to do with elasticsearch_master being a "gateway" where you would expose the ports? Wouldn't it be better to have HA running and reverse proxy to the elastic masters?

The reason we have 3 variants is for flexibility in deploying the 3 roles. In a small-to-middling-size environment one would just deploy all the ES instances as master+data. In a large production environment typically you'll want to send all the writes thru masters and do all the reads on data-only instances. We didn't want to be overly opinionated about the topology in this blueprint.